### PR TITLE
ARC4 semantic issues workarounds

### PR DESCRIPTION
**Pull request by achidlow**

### CODE REVIEW

File: tests/test_execution.py [hunk 1]
```diff
@@ -1333,7 +1334,50 @@ def clear_state_program(self) -> bool:
                 return True
 
     # expected error message here is just an example, update test once a better one is available
-    import puya.errors
-
     with pytest.raises(puya.errors.CodeError, match="Invalid use of a type union"):
         harness.deploy_from_closure(test)
+
+
+def test_mutable_ref_param_reassignment_fails(harness: _TestHarness) -> None:
```
Comment (daniel-makerx) [inline tests/test_execution.py:1341]: These would be better as part of the expected_errors.test suite?

File: tests/test_execution.py [hunk 2]
```diff
@@ -1333,7 +1334,50 @@ def clear_state_program(self) -> bool:
...
+    def test() -> None:
+        from puyapy import arc4
+
+        class Baddie(arc4.ARC4Contract):
+            @arc4.abimethod
+            def okay(self, arr: arc4.DynamicBytes) -> None:
+                arr = arc4.DynamicBytes.from_bytes(arr.bytes)
```
Comment (tristanmenzel) [inline tests/test_execution.py:1348]: Probably worth adding a comment explaining _why_ this assignment is ok

**Review by tristanmenzel — APPROVED** [review]

File: tests/test_execution.py (hunk 1 above)
Comment (achidlow) [reply tests/test_execution.py:1341]: Because I thought (wrongly) that those tests didn't lower to IR, where this error is generated.

Comment (daniel-makerx) [reply tests/test_execution.py]: I've moved them as I suggested

**Pull request merged**